avogadro/core/utilities.h

```diff
@@ -104,12 +105,17 @@ inline std::string rstrip(const std::string& str, char c)
 /**
  * @brief Cast the inputString to the specified type.
  * @param inputString String to cast to the specified type.
+ * @retval converted value if cast is successful
+ * @retval std::nullopt otherwise
  */
 template <typename T>
-T lexicalCast(const std::string& inputString)
+std::optional<T> lexicalCast(const std::string& inputString)
```

Comment:
You can check whether cast is successful by `if (o) {}` or `if (o.has_value()) {}`, and obtain the converted value `*o` or `o.value()`.

Reply:
Yes, this seems useful, particularly the range version